src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -702,10 +702,13 @@ protected DynamoModel(IStartConfiguration config)
                     // Do nothing for now
                 }
             }
+            //align disable analytics with nonetwork mode, this is done in other entry points(CLI,Sandbox etc) - but
+            //not all integrators will use those entry points, some may just create a DynamoModel directly.
+            Analytics.DisableAnalytics = NoNetworkMode;
 
             // If user skipped analytics from assembly config, do not try to launch the analytics client
             // or the feature flags client for web traffic reason.
-            if (!IsServiceMode && !areAnalyticsDisabledFromConfig && !Analytics.DisableAnalytics)
+            if (!IsServiceMode && !areAnalyticsDisabledFromConfig && !Analytics.DisableAnalytics && !NoNetworkMode)
```

Comment:
I'm not sure why I skipped this check in https://github.com/DynamoDS/Dynamo/pull/14526 for `nonetworkmode`, perhaps because I thought `DisableAnalytics` already covers that case and is redundant, can't remember.